core/trino-web-ui/src/main/java/io/trino/web/ui/WebUiResources.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.web.ui;
+
+import com.google.common.io.Resources;
+import jakarta.ws.rs.core.Response;
+import org.eclipse.jetty.http.MimeTypes;
+import org.eclipse.jetty.util.FileID;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import static jakarta.ws.rs.core.Response.Status.NOT_FOUND;
+
+public class WebUiResources
+{
+    private static final MimeTypes MIME_TYPES = new MimeTypes();
+
+    private WebUiResources() {}
+
+    public static Response webUiResource(String path)
+            throws IOException
+    {
+        if (!isCanonical(path)) {
+            // consider redirecting to the absolute path
+            return Response.status(NOT_FOUND).build();
+        }
+
+        try {
+            URL resource = Resources.getResource(WebUiResources.class, path);
+            return Response.ok(resource.openStream(), mediaType(resource.toString())).build();
+        }
+        catch (IllegalArgumentException e) {
+            return Response.status(NOT_FOUND).build();
+        }
+    }
+
+    public static String readWebUiResource(String path)
+            throws IOException
+    {
+        return Resources.toString(Resources.getResource(WebUiResources.class, path), StandardCharsets.UTF_8);
+    }
+
+    public static boolean isCanonical(String fullPath)
+    {
+        try {
+            return new URI(fullPath).normalize().getPath().equals(fullPath);
+        }
+        catch (URISyntaxException e) {
+            return false;
+        }
+    }
+
+    private static String mediaType(String filename)
+    {
+        String extension = FileID.getExtension(filename);
```

Comment:
Rather than depending on Jetty, let's just have an inline switch statement here, as we only have a few file extensions. Having the Jetty version in Trino makes upgrades harder and has the potential for mismatches.

Reply:
yep, I had second thought on that as well.